src/DeviceStatusPage/AliasesCard.tsx

```diff
@@ -74,26 +75,34 @@ const AliasesCard = ({
   onNewAliasClick,
   onEditAliasClick,
   onRemoveAliasClick,
-}: AliasesCardProps): React.ReactElement => (
-  <FullHeightCard xs={12} md={6} className="mb-4">
-    <Card.Header as="h5">Aliases</Card.Header>
-    <Card.Body className="d-flex flex-column">
-      {device.aliases.size > 0 ? (
-        <AliasesTable
-          aliases={device.aliases}
-          onEditAliasClick={onEditAliasClick}
-          onRemoveAliasClick={onRemoveAliasClick}
-        />
-      ) : (
-        <p>Device has no aliases</p>
-      )}
-      <div className="mt-auto">
-        <Button variant="primary" onClick={onNewAliasClick} disabled={device.deletionInProgress}>
-          Add alias
-        </Button>
-      </div>
-    </Card.Body>
-  </FullHeightCard>
-);
+}: AliasesCardProps): React.ReactElement => {
+  const astarte = useAstarte();
+  return (
+    <FullHeightCard xs={12} md={6} className="mb-4">
+      <Card.Header as="h5">Aliases</Card.Header>
+      <Card.Body className="d-flex flex-column">
+        {device.aliases.size > 0 ? (
+          <AliasesTable
+            aliases={device.aliases}
+            onEditAliasClick={onEditAliasClick}
+            onRemoveAliasClick={onRemoveAliasClick}
+          />
+        ) : (
+          <p>Device has no aliases</p>
+        )}
+        <div className="mt-auto">
+          <Button
+            variant="primary"
+            onClick={onNewAliasClick}
+            hidden={!astarte.token?.can('appEngine', 'PATCH', `/devices/${device.id}`)}
+            disabled={device.deletionInProgress}
+          >
+            Add alias
+          </Button>
```

Comment:
This check should be run also to know whether to display the actions to edit or remove an alias in the AliasesTable.